### PR TITLE
wapm-cli: 0.5.5 -> 0.5.6

### PR DESCRIPTION
**Pull request by lucperkins**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
@ofborg build wapm-cli

**Comment by SuperSandro2000** [comment]
```
error: failed to run custom build command for `openssl-sys v0.9.75`
Caused by:
  process didn't exit successfully: `/build/source/target/release/build/openssl-sys-7c5125cfaf0a8fc4/build-script-main` (exit status: 101)
  --- stdout
  cargo:rustc-cfg=const_fn
  cargo:rerun-if-env-changed=X86_64_UNKNOWN_LINUX_GNU_OPENSSL_NO_VENDOR
  X86_64_UNKNOWN_LINUX_GNU_OPENSSL_NO_VENDOR unset
  cargo:rerun-if-env-changed=OPENSSL_NO_VENDOR
  OPENSSL_NO_VENDOR unset
  CC_x86_64-unknown-linux-gnu = Some("/nix/store/0vln6gxc9b2wr9yhbn0aspy6badlj638-gcc-wrapper-11.3.0/bin/cc")
  CFLAGS_x86_64-unknown-linux-gnu = None
  CFLAGS_x86_64_unknown_linux_gnu = None
  HOST_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  running "perl" "./Configure" "--prefix=/build/source/target/x86_64-unknown-linux-gnu/release/build/openssl-sys-7fdb9e2ad6f4a942/out/openssl-build/install" "--openssldir=/usr/local/ssl" "no-dso" "no-shared" "no-ssl3" "no-unit-test" "no-comp" "no-zlib" "no-zlib-dynamic" "no-md2" "no-rc5" "no-weak-ssl-ciphers" "no-camellia" "no-idea" "no-seed" "linux-x86_64" "-O2" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64"
  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /build/wapm-cli-0.5.6-vendor.tar.gz/openssl-src/src/lib.rs:488:39
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: builder for '/nix/store/6an9vipd62fmsca9pw966ai3v6wbwgmc-wapm-cli-0.5.6.drv' failed with exit code 101;
```

**Comment by lucperkins** [comment]
@SuperSandro2000 I can't test on Linux, I'm afraid, but AFAIK this `pkg-config` dependency should resolve this.

File: pkgs/tools/package-management/wapm/cli/default.nix [hunk 1]
```diff
@@ -10,19 +11,20 @@
...
     repo = "wapm-cli";
     rev = "v${version}";
-    sha256 = "sha256-BKBd1tJwV4VOjRnAx/spQy3LIXzujrO2SS5eA1uybNA=";
+    sha256 = "sha256-QgQQ0lbr7Ggd2HmKlu/5TMFISxrwKWstoAq8e776l8I=";
   };
 
-  cargoSha256 = "sha256-dv04AXOnzizjq/qx3qy524ylQHgE4gIBgeYI+2IRTug=";
+  cargoSha256 = "sha256-F4BvP1yMJ06QgocD6InwkfHtJIsEkOfxuaaalksJCew=";
 
   buildInputs = [ libiconv openssl ]
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/wapm/cli/default.nix:25]:
```suggestion
  buildInputs = lib.optionals stdenv.isLinux [ openssl pkg-config ]
```
or is openssl used on darwin, too? That is not always the case.

File: pkgs/tools/package-management/wapm/cli/default.nix [hunk 2]
```diff
@@ -10,19 +11,19 @@
...
-    sha256 = "sha256-BKBd1tJwV4VOjRnAx/spQy3LIXzujrO2SS5eA1uybNA=";
+    sha256 = "sha256-QgQQ0lbr7Ggd2HmKlu/5TMFISxrwKWstoAq8e776l8I=";
   };
 
-  cargoSha256 = "sha256-dv04AXOnzizjq/qx3qy524ylQHgE4gIBgeYI+2IRTug=";
+  cargoSha256 = "sha256-F4BvP1yMJ06QgocD6InwkfHtJIsEkOfxuaaalksJCew=";
 
-  buildInputs = [ libiconv openssl ]
-    ++ lib.optionals stdenv.isDarwin [ Security SystemConfiguration ];
+  buildInputs = lib.optionals stdenv.isLinux [ openssl pkg-config ]
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/wapm/cli/default.nix:25]: 🤦🏼 pkg-config in buildInputs. What was I thinking.

```suggestion
  buildInputs = lib.optionals stdenv.isLinux [ openssl ]
```

File: pkgs/tools/package-management/wapm/cli/default.nix [hunk 3]
```diff
@@ -10,19 +11,19 @@
...
   src = fetchFromGitHub {
     owner = "wasmerio";
     repo = "wapm-cli";
     rev = "v${version}";
-    sha256 = "sha256-BKBd1tJwV4VOjRnAx/spQy3LIXzujrO2SS5eA1uybNA=";
+    sha256 = "sha256-QgQQ0lbr7Ggd2HmKlu/5TMFISxrwKWstoAq8e776l8I=";
   };
 
-  cargoSha256 = "sha256-dv04AXOnzizjq/qx3qy524ylQHgE4gIBgeYI+2IRTug=";
+  cargoSha256 = "sha256-F4BvP1yMJ06QgocD6InwkfHtJIsEkOfxuaaalksJCew=";
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/wapm/cli/default.nix:24]:
```suggestion

  nativeBuildInputs = lib.optionals stdenv.isLinux [ pkg-config ]

```

**Review by SuperSandro2000 — COMMENTED** [review]
sorry about that

File: pkgs/tools/package-management/wapm/cli/default.nix [hunk 4]
```diff
@@ -10,19 +11,21 @@
...
 
-  buildInputs = [ libiconv openssl ]
-    ++ lib.optionals stdenv.isDarwin [ Security SystemConfiguration ];
+  nativeBuildInputs = lib.optionals stdenv.isLinux [ pkg-config ]
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/wapm/cli/default.nix:25]:
```suggestion
  nativeBuildInputs = lib.optionals stdenv.isLinux [ pkg-config ];
```

**Comment by SuperSandro2000** [comment]
@ofborg build wapm-cli

**Comment by SuperSandro2000** [comment]
and fails to build:

```
  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /build/wapm-cli-0.5.6-vendor.tar.gz/openssl-src/src/lib.rs:488:39
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: builder for '/nix/store/yj6lj5hxiczgcsrq2wmf8dschl3arrrf-wapm-cli-0.5.6.drv' failed with exit code 101;
```

openssl-sys missing?

**Comment by Artturin** [comment]
We moved past this in https://github.com/NixOS/nixpkgs/pull/203742
And it built?

**Pull request closed without merging**